src/app/projects/pages/autofill-project-form/autofill-project-form.component.ts

```diff
@@ -37,100 +38,65 @@ export class AutofillProjectFormComponent implements OnInit {
       Validators.required,
       Validators.pattern(/^10.\d{4,9}\/[-._;()\/:A-Z0-9]+$/i)
     ]));
+    this.geoAccessionCtrl = new FormControl('', Validators.compose([
```

Comment:
This is for sure much better but I was more thinking to separate out the components themselves so the UI is separate too using [EventEmitters](https://angular.io/guide/component-interaction#parent-listens-for-child-event). This is still good SoC to I will approve and it's up to you guys!

Reply:
yeah, we thought of doing it initially but I think the component which contains the form with 2 fields are still acceptable, so we just separate the geo & doi handlers into their own services.